p_sets/3/find.rb

```diff
@@ -1,31 +1,32 @@
 module Finder
   def self.linear_search(needle, hay)
-    max_index   = hay.length - 1
+    max_index   = hay.length > 1 ? hay.length - 1 : 1
     result      = false
     max_index.times do |i|
-      result = true if hay[i] == needle
+      if hay[i] == needle
+        result = true
+        break
+      end
     end
-    puts result.to_s
+    result
   end
```

Comment:
This'll still fail if you do `linear_search(1, [0,1])`. The problem wasn't with lengths of 1 specifically but with the needle being the last element.